app/models/policy.rb

```diff
@@ -55,9 +55,8 @@ def initial_profile
 
   def destroy_orphaned_business_objective
     bo_changes = (previous_changes.fetch(:business_objective_id, []) +
-                  changes.fetch(:business_objective_id, [])).compact
-    return if bo_changes.blank?
```

Comment:
When you destroy a policy, this `bo_changes` is empty; that's the crux of the bug. Now we include the `business_objective_id` in the collection as well.